src/Hl7.Fhir.Base/ElementModel/TypedElementOnSourceNode.cs

```diff
@@ -45,7 +48,7 @@ public TypedElementOnSourceNode(ISourceNode source, string type, IStructureDefin
                     throw Error.Format(nameof(type), $"Cannot determine the type of the root element at '{_source.Location}', " +
                         $"please supply a type argument.");
                 else
-                    return (rootType, null);
+                    return (rootType!, null);
```

Comment:
Let's consider this for a bit - for the first time we are kind of lying here, and we're now saying: "in 98% of the cases, this is non-null - so it's not null".  But is 98% good enough?  Or should we drop de bang and add a #pragma to make it clear we're dodging the nullability warning?